dev/Pager/TestUI/PagerPage.xaml.cs

```diff
@@ -37,32 +37,73 @@ private void OnLoad(object sender, RoutedEventArgs args)
             NextPageButtonVisibilityComboBox.SelectionChanged += OnNextButtonVisibilityChanged;
             LastPageButtonVisibilityComboBox.SelectionChanged += OnLastButtonVisibilityChanged;
 
-            NumberBoxVisibilityTextBlock.Text = TestPager.NumberBoxDisplayTestHook.Visibility == Visibility.Collapsed ? "Collapsed" : "Visible";
-            ComboBoxVisibilityTextBlock.Text = TestPager.ComboBoxDisplayTestHook.Visibility == Visibility.Collapsed ? "Collapsed" : "Visible";
+            TestPager.NumberPanelDisplayTestHook.ElementPrepared += OnElementPrepared;
+
+         
+            NumberBoxVisibilityCheckBox.IsChecked = TestPager.NumberBoxDisplayTestHook.Visibility == Visibility.Visible;
+            ComboBoxVisibilityCheckBox.IsChecked = TestPager.ComboBoxDisplayTestHook.Visibility == Visibility.Visible;
+            NumberPanelVisibilityCheckBox.IsChecked = TestPager.NumberPanelDisplayTestHook.Visibility == Visibility.Visible;
             NumberBoxIsEnabledCheckBox.IsChecked = TestPager.NumberBoxDisplayTestHook.IsEnabled;
             ComboBoxIsEnabledCheckBox.IsChecked = TestPager.ComboBoxDisplayTestHook.IsEnabled;
+            UpdateNumberPanelContentTextBlock(this, null);
+        }
+
+        private void OnElementPrepared(ItemsRepeater sender, ItemsRepeaterElementPreparedEventArgs args)
+        {
+            var element = args.Element as FrameworkElement;
```

Comment:
What is the purpose of this code?

Reply:
When I was testing out the Number Panel, there was no easy way for me to differentiate between the First,Prev,Next,Last buttons and the buttons within the repeater. I thought the easiest way to help me identify which buttons belong to the itemsrepeater was to set the name on each element when they are being prepared and search for those names in the tests.